api/envoy/config/route/v3/route_components.proto

```diff
@@ -696,9 +697,30 @@ message RouteAction {
 
     reserved "runtime_key";
 
+    // Only one of *cluster* and *cluster_header* can be specified.
```

Comment:
Unfortunately when `cluster_header` is hidden, then this comment doesn't make much sense (see https://storage.googleapis.com/envoy-pr/2ce6b19/docs/api-v3/config/route/v3/route_components.proto.html#config-route-v3-routeaction-requestmirrorpolicy).
Is there much left to implement this feature? If not, we can maybe implement it as part of this PR.

Reply:
Not too much, but many are from Tests. In that case I can continue to commit to this PR. At least API is reviewed now.